File: School.Domain/Entities/Teacher.cs
using System;
using System.Collections.Generic;

namespace School.Domain.Entities
{
    public class Teacher
    {
        public int Id { get; set; }
        public string FullName { get; set; }
        public string MaritalStatus { get; set; }
        public DateTime Birthday { get; set; }
        public int ContactId { get; set; }
        public Contact Contact { get; set; }
        public int AddressId { get; set; }
        public Address Address { get; set; }
        public List<DisciplineTeacher> Disciplines {get; set;}= new List<DisciplineTeacher>();

    }

}

File: Views/src/app/organization/interfaces/Teacher.ts
import { DisciplineTeacher } from './DisciplineTeacher';
import { Address } from './Address';
import { Contact } from './Contact';

export class Teacher {
    constructor() {

    }

    id: number;
    fullName: string;
    birthday: Date;
    maritalstatus: string;
    contactid: number;
    contact: Contact;
    disciplines: DisciplineTeacher[];
    addressid: number;
    address: Address;
}
